File: MVVMJavaRx/app/src/main/java/com/vascome/fogtail/presentation/base/router/BaseRouter.java
package com.vascome.fogtail.presentation.base.router;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import com.vascome.fogtail.presentation.base.activity.BaseActivity;

/**
 * Created by vasilypopov on 12/12/17
 * Copyright (c) 2017 CleanJava. All rights reserved.
 */

abstract public class BaseRouter<A extends BaseActivity> {
    private A activity;
    private FragmentManager fragmentManager;

    public BaseRouter(A activity) {
        this.activity = activity;
        this.fragmentManager = activity.getSupportFragmentManager();
    }


    public void replaceFragment(int content, Fragment fragment) {

        fragmentManager.beginTransaction()
                .replace(content, fragment)
                .commit();
    }

    public A getActivity() {
        return activity;
    }
}

File: MVVMKotlin/app/src/main/java/com/vascome/fogtail/presentation/base/router/BaseRouter.kt
package com.vascome.fogtail.presentation.base.router

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager

import com.vascome.fogtail.presentation.base.activity.BaseActivity

/**
 * Created by vasilypopov on 12/12/17
 * Copyright (c) 2017 CleanJava. All rights reserved.
 */

abstract class BaseRouter<out A : BaseActivity>(val activity: A) {
    private val fragmentManager: FragmentManager = activity.supportFragmentManager

    fun replaceFragment(content: Int, fragment: Fragment) {

        fragmentManager.beginTransaction()
                .replace(content, fragment)
                .commit()
    }
}
